connect/api/src/test/java/org/apache/kafka/connect/sink/SinkRecordTest.java

```diff
@@ -125,4 +125,19 @@ public void shouldModifyRecordHeader() {
         Header header = record.headers().lastWithName("intHeader");
         assertEquals(100, (int) Values.convertToInteger(header.schema(), header.value()));
     }
-}
\ No newline at end of file
+
+    @Test
+    public void shouldRetainOriginalTopicPartition() {
```

Comment:
Can we update the declaration for `record` in the `beforeEach` method to explicitly use the constructor that takes in the original TPO? Right now we're implicitly relying on that behavior with this test.

If we want to be paranoid (we don't have to but it's up to you), we can also add a test case for the older constructors that verifies that they provide the possibly-transformed TPO as the original TPO as well.